### ISSUE TITLE
Calliptaminae in North Africa

### ISSUE THREAD

**Issue by klausriede** [sfg-taxonpages/orthoptera#7]
![grafik](https://github.com/sfg-taxonpages/orthoptera/assets/12816652/c1bc1814-6f02-4e5c-9642-b9d42ec1ed09)

hmm. of course there are Calliptaminae in this area...maybe no specimens in the database? How is geosearch made? It should be an intersection with distribution AREAS ie shapes not searching for specimens

**Comment by mjy** [comment]
I'm curious, why should a spatial search not reference both specimens and AssertedDistributions (which are actually abstractions of specimens)?

**Comment by mjy** [comment]
As for the square- the search presently only returns results if area that is tied to the data is completely enclosed. Partial intersections are not returned.

**Comment by klausriede** [comment]
which does not make sense from a biological point of view...I want to see the species possibly occuring in this square

**Comment by mjy** [comment]
I don't think this has anything to do with Biology, does it?  It's the difference between selecting objects in Illustrator and Powerpoint, the selection is a process, an interaction with an interface.  The outcome depends on the search mechanism.

**Comment by klausriede** [comment]
ok let us talk about the search algorithm. We have a search bounding box and object bounding boxes and points (from species distributions). If we search only objects (species..) totally INSIDE the search box we get endemics for this area . If we search for overlaps we will find all species which MIGHT occur during a field trip within this area (which I suppose is the  more frequent "biological" question). Both geoqueries are relevant, but should be indicated somehow. Tell us more about the algorithm!

**Comment by typophyllum** [comment]
The real problem seems to be the asserted distribution in southern Chile. This is based on genus _Calliptamus_, but none of its 17 species occurs there. Where does this come from? (The distribution in French Guiana retaed too its distribution in France is a known problem.)

**Comment by mabecabrera** [comment]
Chile distribution solved! Thaks to @mjy. Very tricky bug!!!
Should we rename this issue? Map search outcome?

**Comment by mjy** [comment]
Issue was not bad data, it was bad shape for SE Chile (wonder if its resolve in Natural Earth gazetter now).

**Comment by klausriede** [comment]
the south chile bug is resolved, but not my original question...

**Comment by mjy** [comment]
> the search presently only returns results if area that is tied to the data is completely enclosed. Partial intersections are not returned.

**Comment by klausriede** [comment]
which I think does not make sense for the common user. Think of it in  Admin terms: if you search for Germany then I want a list of species occuring in Germany. The other option would be endemisms from Germany (probably zero) . Also an interesting query, the two options would be good, but should be indicated!

**Comment by mjy** [comment]
> which I think does not make sense for the common user

The pattern being used here is used in countless software packages where objects in an interface are selected. In those packages the user must completely surround the target for it to become selected.  One has to either learn that this is the method or that the intersection based-approach is.  Either way, one must learn, so process/method wise this is objectively equal amount of work to be done by the person figuring out what is going on.  I personally feel that references to a "common user" tend to be "True Scottsman" in nature, they don't really help to further things along in my experience.  

Each panel in TaxonPages seeks to answer a question. This map panel, as originally conceived, seeks to only answer the question "where, generally in the world, is this taxon found", that's it.  It succeeds, I think you would agree, at this first step.  A myriad of other questions exist, whether they all should be answered in the same panel is the discussion.  Nearly all questions I can think of can be answered via the data in the DwC export.

This does, however re-iterate the need for documenting the intent and functionality for each panel.  We understand this need.

**Comment by klausriede** [comment]
so let me illustrate my needs again, using the inaturalist search panel https://www.inaturalist.org/observations?nelat=32.35861937757444&nelng=35.28533935546875&place_id=any&subview=map&swlat=23.435025232708952&swlng=28.38592529296875&taxon_id=510317
![grafik](https://github.com/sfg-taxonpages/orthoptera/assets/12816652/c9267ec5-3f86-4720-b96f-9be987a5ee73)

with a similar bounding box searching for Calliptaminae, resulting in 2 species. I think considering asserted distributions there are many more to be expected in the area (maybe a collector wants to travel there and find more specimens), and for me as a user this is what I would expect. Would love to hear opinions from Orthopterists!

**Comment by mjy** [comment]
Thanks for the example.

Curious that iNaturalist has the same behaviour as we do, only objects inside the box are returned.  Might it not be confusing to have CollectionObjects need to be in the box, but AssertedDistributions clipping/intersecting, particularly since CollectionObjects can have shape based georeferences in TaxonWorks? 

What is the behaviour for boxes clipping a state that has point data that are not in the box, are those data supposed to be included by extension because people will see say Texas asserted distributions, but not all points in Texas.  I find a myriad of questions like this present just as much potential confusion not.

Note that you can use non-rectangular polygons to search on TaxonPages and in TW, it doesn't take much to wrap the geographic areas in question.

**Comment by klausriede** [comment]
Yes inaturalist only has points, it does not work with shapes. So this is correct. And in fact the Egypt data from inaturalist are totally insufficient, reflecting bad coverage. This is the advantage of OSF, and it should be maintained in TP

**Comment by klausriede** [comment]
No, search was for geographical units. for example Great Britain http://orthoptera.speciesfile.org/Common/editTaxon/SearchForTaxon.aspx gives you a list of extant species not exclusively existing in Britain. Interestingly it does not work for scotland.

**Comment by mjy** [comment]
> This is the advantage of OSF, and it should be maintained in TP
...
> gives you a list of extant species not exclusively existing in Britain.

The search in TaxonPages gives you a list of species (we don't check extant) not exclusively found in that spatial area provided. It does so with more potential power, using a true spatial query (instead of a lookup which selects a value that must be matched against) on both specimens and asserted distributions.   

In addition we provide the DwC that lets you do exactly what you did before, again with more power and resolution, you can download the DwC and filter on the value (as is done in Species File) "Great Britian".  [Open refine](https://openrefine.org/) is your friend here. Just like in OSF there are going to be exceptions to the data.


I feel that we are picking at straws.
